Code/Framework/AzCore/Tests/DOM/DomJsonBenchmarks.cpp

```diff
@@ -0,0 +1,184 @@
+        ->Args({ 10, 5 })                                                                                                                  \
+        ->Args({ 10, 500 })                                                                                                                \
+        ->Args({ 100, 5 })                                                                                                                 \
+        ->Args({ 100, 500 })                                                                                                               \
+        ->Unit(benchmark::kMillisecond);
+
+    BENCHMARK_DEFINE_F(DomJsonBenchmark, DomDeserializeToDocumentInPlace)(benchmark::State& state)
+    {
+        AZ::DOM::JsonBackend backend;
+        AZStd::string serializedPayload = GenerateDomJsonBenchmarkPayload(state.range(0), state.range(1));
+
+        for (auto _ : state)
+        {
+            state.PauseTiming();
+            AZStd::string payloadCopy = serializedPayload;
+            state.ResumeTiming();
+
+            auto result = AZ::DOM::Json::WriteToRapidJsonDocument(
+                [&](AZ::DOM::Visitor* visitor)
+                {
+                    return backend.ReadFromStringInPlace(payloadCopy, visitor);
+                });
+
+            benchmark::DoNotOptimize(result.GetValue());
+        }
+
+        state.SetBytesProcessed(serializedPayload.size() * state.iterations());
+    }
+    BENCHMARK_REGISTER_JSON(DomJsonBenchmark, DomDeserializeToDocumentInPlace)
+
+    BENCHMARK_DEFINE_F(DomJsonBenchmark, DomDeserializeToDocument)(benchmark::State& state)
+    {
+        AZ::DOM::JsonBackend backend;
+        AZStd::string serializedPayload = GenerateDomJsonBenchmarkPayload(state.range(0), state.range(1));
+
+        for (auto _ : state)
+        {
+            auto result = AZ::DOM::Json::WriteToRapidJsonDocument(
+                [&](AZ::DOM::Visitor* visitor)
+                {
+                    return backend.ReadFromString(serializedPayload, AZ::DOM::Lifetime::Temporary, visitor);
+                });
+
+            benchmark::DoNotOptimize(result.GetValue());
+        }
+
+        state.SetBytesProcessed(serializedPayload.size() * state.iterations());
+    }
+    BENCHMARK_REGISTER_JSON(DomJsonBenchmark, DomDeserializeToDocument)
+
+    BENCHMARK_DEFINE_F(DomJsonBenchmark, JsonUtilsDeserializeToDocument)(benchmark::State& state)
+    {
+        AZ::DOM::JsonBackend backend;
+        AZStd::string serializedPayload = GenerateDomJsonBenchmarkPayload(state.range(0), state.range(1));
+
+        for (auto _ : state)
+        {
+            auto result = AZ::JsonSerializationUtils::ReadJsonString(serializedPayload);
+
+            benchmark::DoNotOptimize(result.GetValue());
+        }
+
+        state.SetBytesProcessed(serializedPayload.size() * state.iterations());
+    }
+    BENCHMARK_REGISTER_JSON(DomJsonBenchmark, JsonUtilsDeserializeToDocument)
+
+#undef BENCHMARK_REGISTER_JSON
+} // namespace Benchmark
```

Comment:
Are there any benchmarks that do the same but directly through RapidJSON for comparison?

Reply:
Yes, `JsonUtilsDeserializeToDocument` goes directly through RapidJSON with the same data-set.